### PR TITLE
DE45868: FACE > Update LTI framing options

### PR DESCRIPTION
**Pull request by coffeexcode**
## Relevant Rally
- [DE45868](https://rally1.rallydev.com/#/?detail=/defect/613300258601&fdp=true): Lessons / New Content Experience > 20.21.10 > Adding External Learning Tools default Display Option for tools that cannot be embedded

## Description
Lessons / Classic content can change the embed status of an LTI link even if the link cannot be embedded. This means that when FACE is on, they would be unable to change the embed status of an LTI link using the FACE page (would have to disable FACE and use lessons/classic). 

With this change, the embed options for LTI will now always display, but if the LTI cannot be embedded, the options to select embed will be disabled. For any existing LTI links that are currently embedded that can't be, will now have the option to change the embed status from the FACE page (and an error will be shown if they try to save the page without changing the embed status).

## Screenshots
New:
Disabled option when embedding is not allowed
image

Error when trying to save without changing embed option
image

### CODE REVIEW

**Comment by prajapac** [comment]
Unit Tests seems to be failing.

**Comment by coffeexcode** [comment]
> Unit Tests seems to be failing.

I think it's CI. It passed the first time and failed on sauce labs for an import error. Looks like other recent PRs are getting the same treatment.

File: components/d2l-activity-editor/d2l-activity-content-editor/lti-link/d2l-activity-content-lti-link-options.js [hunk 1]
```diff
@@ -103,7 +111,8 @@ class ContentEditorLtiLinkOptions extends SkeletonMixin(ErrorHandlingMixin(Local
 					name="link-display-group"
 					value="newTab"
 					?checked="${isExternalResource}"
-					@change="${this._saveLinkOptions}">
+					@change="${this._saveLinkOptions}"
+					aria-invalid="${this._linkOptionsError ? 'true' : 'false'}">
```
Comment (prajapac) [inline components/d2l-activity-editor/d2l-activity-content-editor/lti-link/d2l-activity-content-lti-link-options.js:114]: What is this property responsible for (more curious as to why both radio options have it 🤔)?

Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/lti-link/d2l-activity-content-lti-link-options.js:114]: aria-invalid is for the save flow the recognize that something is invalid. Radio buttons operate essentially as a group (only one from the entire group can be selected) so the property exists on both to show the whole group as invalid.

**Review by prajapac — APPROVED** [review]

File: components/d2l-activity-editor/d2l-activity-content-editor/lti-link/d2l-activity-content-lti-link-options.js [hunk 2]
```diff
@@ -79,21 +72,35 @@ class ContentEditorLtiLinkOptions extends SkeletonMixin(ErrorHandlingMixin(Local
...
 				<input
 					id="embed-on-page"
 					type="radio"
 					name="link-display-group"
 					value="embed"
 					?checked="${!isExternalResource}"
-					@change="${this._saveLinkOptions}">
+					?disabled="${!this.canEmbed}"
+					@change="${this._saveLinkOptions}"
+					aria-invalid="${this._linkOptionsError ? 'true' : 'false'}">
```
Comment (brissons) [inline components/d2l-activity-editor/d2l-activity-content-editor/lti-link/d2l-activity-content-lti-link-options.js:103]: can this be simplified to `!!this._linkOptionsError`?

File: components/d2l-activity-editor/d2l-activity-content-editor/lti-link/d2l-activity-content-lti-link-options.js [hunk 3]
```diff
@@ -79,21 +72,35 @@ class ContentEditorLtiLinkOptions extends SkeletonMixin(ErrorHandlingMixin(Local
...
+		}
+
+		this.clearError('_linkOptionsError');
+		this.setError('_linkOptionsError', error, 'link-options-tooltip');
+	}
+
 	_renderLinkOptions(isExternalResource) {
 		return html`
+			${this._renderLinkOptionsTooltip()}
 			<label class="d2l-input-radio-label d2l-skeletize">
```
Comment (brissons) [inline components/d2l-activity-editor/d2l-activity-content-editor/lti-link/d2l-activity-content-lti-link-options.js:94]: I think the label also has a class we can use if it's disabled: `d2l-input-radio-label-disabled`

**Review by brissons — COMMENTED** [review]
lgtm! just a couple comments:

**Review by brissons — APPROVED** [review]

File: components/d2l-activity-editor/d2l-activity-content-editor/lti-link/d2l-activity-content-lti-link-options.js (hunk 3 above)
Comment (coffeexcode) [reply components/d2l-activity-editor/d2l-activity-content-editor/lti-link/d2l-activity-content-lti-link-options.js:94]: Nice! Added that in

**Review by brissons — APPROVED** [review]

**Pull request merged**